### PR TITLE
Activiti/Activiti#3460-Added application deployed event producer

### PR DESCRIPTION
**Pull request by Giovanni007**
Created the producer for the event application deployed. The goal is to generate the event that we can handle in Query to get information about the application and create an entity there. (we need it to return all the versions of an Application using an API)

### CODE REVIEW

File: activiti-core/activiti-spring-boot-starter/src/test/java/org/activiti/spring/boot/DeployedApplicationListener.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    
+    @Override
+    public void onEvent(ApplicationDeployedEvent event) {
+            deployedApplications.add(event.getEntity()); 
+    }
+    
+    public List<Deployment> getDeployedApplication() {
+            return deployedApplications; 
+    }
+}
```
Comment (andrea-ligios) [inline activiti-core/activiti-spring-boot-starter/src/test/java/org/activiti/spring/boot/DeployedApplicationListener.java]: Please add a newline character at line 38

File: activiti-core/activiti-spring-boot-starter/src/main/java/org/activiti/spring/ApplicationDeployedEventProducer.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+                for (ProcessRuntimeEventListener<ApplicationDeployedEvent> listener : listeners) {
+                    listener.onEvent(applicationDeployedEvent);
+                }
+            }
+        }
+        return applicationDeployedEvents;
+    }
+
+    @Override
+    public void doStop() {
```
Comment (andrea-ligios) [inline activiti-core/activiti-spring-boot-starter/src/main/java/org/activiti/spring/ApplicationDeployedEventProducer.java:73]: Please add a comment inside the empty method body to make Codacy happy

**Review by andrea-ligios — CHANGES_REQUESTED** [review]
Everything seems fine apart from two very minor additions needed.

I'm not sure if this build problem is ephemeral, and hence a false positive: https://lgtm.com/projects/g/Activiti/Activiti/logs/rev/pr-bb6447d5c9c96f4b7024e584c6860c90772ea837/lang:java/stage:Build%20merge_53a3a4ce0250a1a344bfab08e2738a9740ce0fec

**Review by andrea-ligios — APPROVED** [review]

File: activiti-core/activiti-spring-boot-starter/src/main/java/org/activiti/spring/ApplicationDeployedEventProducer.java [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+import org.activiti.runtime.api.model.impl.APIDeploymentConverter;
+import org.springframework.context.ApplicationEventPublisher;
+
+public class ApplicationDeployedEventProducer extends AbstractActivitiSmartLifeCycle {
+
+    private RepositoryService repositoryService;
+    private APIDeploymentConverter deploymentConverter;
+    private List<ProcessRuntimeEventListener<ApplicationDeployedEvent>> listeners;
+    private ApplicationEventPublisher eventPublisher;
+    private static final String APPLICATION_DEPLOYED_EVENT_NAME= "SpringAutoDeployment";
```
Comment (erdemedeiros) [inline activiti-core/activiti-spring-boot-starter/src/main/java/org/activiti/spring/ApplicationDeployedEventProducer.java]: I'm wondering if APPLICATION_DEPLOYMENT_NAME wouldn't be a better name for this constant. What do you think?

File: activiti-core/activiti-spring-boot-starter/src/main/java/org/activiti/spring/ApplicationDeployedEventProducer.java [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+    public void doStart() {
+        List<ApplicationDeployedEvent> applicationDeployedEvents = getApplicationDeployedEvents();
+        if (!applicationDeployedEvents.isEmpty()) {
+            eventPublisher.publishEvent(new ApplicationDeployedEvents(applicationDeployedEvents));
+        }
+    }
+
+    private List<ApplicationDeployedEvent> getApplicationDeployedEvents() {
+        List<Deployment> deployments = deploymentConverter.from(repositoryService
+                        .createDeploymentQuery()
```
Comment (erdemedeiros) [inline activiti-core/activiti-spring-boot-starter/src/main/java/org/activiti/spring/ApplicationDeployedEventProducer.java:60]: The deployment name can be filtered directly here and the `if` in the line `62` can be removed:
`.deploymentName()`.

File: activiti-core/activiti-spring-boot-starter/src/main/java/org/activiti/spring/ApplicationDeployedEventProducer.java [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+
+    @Override
+    public void doStart() {
+        List<ApplicationDeployedEvent> applicationDeployedEvents = getApplicationDeployedEvents();
+        if (!applicationDeployedEvents.isEmpty()) {
+            eventPublisher.publishEvent(new ApplicationDeployedEvents(applicationDeployedEvents));
+        }
+    }
+
+    private List<ApplicationDeployedEvent> getApplicationDeployedEvents() {
```
Comment (erdemedeiros) [inline activiti-core/activiti-spring-boot-starter/src/main/java/org/activiti/spring/ApplicationDeployedEventProducer.java:58]: I think you need either rename this method or remove the call to listeners from it: it's doing more than getting events; it's also calling listeners on them.

**Review by erdemedeiros — CHANGES_REQUESTED** [review]
Some minor details. LGTM otherwise.

File: activiti-core/activiti-spring-boot-starter/src/main/java/org/activiti/spring/ApplicationDeployedEventProducer.java (hunk 4 above)
Comment (Giovanni007) [reply activiti-core/activiti-spring-boot-starter/src/main/java/org/activiti/spring/ApplicationDeployedEventProducer.java:60]: good catch, thanks!

**Review by erdemedeiros — APPROVED** [review]
LGTM

**Pull request merged**